### ISSUE TITLE
Question: Multiple compose backstacks supported

### ISSUE THREAD

**Issue by pandasys** [Zhuinden/simple-stack-compose-integration#13]
I don't see any examples regarding this. Is simple-stack-compose-integration designed with multiple compose backstack support? For example, I'd like an outer backstack closer to the activity level and I'd also like a backstack contained within a BottomSheetScaffold. Before I spend time on a sizeable prototype I was hoping you could either say  yes, no, or maybe yes with caveats. This will be greenfield development, no fragments, views, etc.

**Comment by Zhuinden** [comment]
Hey, I theoretically like the idea, and there is a possible concept for multi-stacks [**using fragments**](https://github.com/Zhuinden/simple-stack/tree/dbd1b1fd12ac20e10c01b6d8be03ade60ccc3868/samples/multistack-samples/simple-stack-example-multistack-nested-fragment/src/main/java/com/zhuinden/simplestackbottomnavfragmentexample/core/navigation), however because of how `BackHandler` works in Compose, I can't think of a way to intercept back in a reliable way to dispatch back to the children and bubble it up to the parent backstack when intercepted by the Composable, while the alternative would be to host Backstacks purely in scoped services, but there I can't think of a way to easily detect which composable is showing and which one isn't (`DisposableEffect(Unit) { onDispose {} }`) and intercept onResume/onPause/onDestroy (which is [exactly what Fragments are used for in that sample](https://github.com/Zhuinden/simple-stack/blob/dbd1b1fd12ac20e10c01b6d8be03ade60ccc3868/samples/multistack-samples/simple-stack-example-multistack-nested-fragment/src/main/java/com/zhuinden/simplestackbottomnavfragmentexample/core/navigation/FragmentStackHostFragment.kt#L31-L57)).

--> So without fragments, I'm not sure how to do it.  <--

I don't think there's an easy way to do it even if I were to write something like Navigation-Compose's `NavBackStackEntry` management (which only works because they pretend that you can always know if a subscreen wants to handle back ahead of time, and can't opt to conditionally intercept back and bubble up if not) which uses `LifecycleRegistry` etc. because even with that, the problems about back event dispatch + detecting lifecycles of both the enclosing context and the navigation scope and the shown composable are a bit heavily tangled.

~~There's a good chance that another "Backstack-looking but not Backstack" that only does navigation but not scoped services/lifecycle management would be needed as a concept to handle this, but that concept doesn't exist, nor is it easy to add (technically what we're looking for is `NavigationCore` but without saved state persistence, but at that rate what's even the point).~~

TL;DR i thought about it and didn't find a good way to do it 🤔 I've generally been using `Backstack` only at top level, and managed subscreen states by other means.

**Comment by Zhuinden** [comment]
An interesting theoretical challenge would be to see if approaches from https://github.com/zsoltk/compose-router can be combined with Simple-Stack to get the power of both 🤔

**Comment by pandasys** [comment]
OK, thanks. I don't have time to help explore this at the moment. I greatly prefer the simple-stack-compose-integration interface to others, especially the Service design. I looked at compose-router but it seemed more proof of concept and generally lacking compared to simple stack.  So, I'm thinking I'll quickly explore using within BottomSheetScaffold and achieve what I need through flags during composition (not showing sheet content, etc). Navigation-Compose is last resort.

**Comment by Zhuinden** [comment]
> . So, I'm thinking I'll quickly explore using within BottomSheetScaffold and achieve what I need through flags during composition (not showing sheet content, etc).

Yeah, that's generally what I did within a screen too.

-----

Nesting of backstacks is a very hard problem. Sometimes I think about it and then think "if I ever do this, will it just make the API confusing"

**Comment by pandasys** [comment]
I pushed my ```composeStateChanger.RenderScreen()``` down into the content of a BottomSheetScaffold, handling hiding/showing other UI components via flags. It's working nicely. I can envision scenarios where nesting backstacks would be advantageous, but for this particular app I can design so one will suffice (at least to this point). Best part is I can avoid the Jetpack Navigation component. Thinking about it a bit, the "limitation" of only one backstack may drive toward a better overall design. Time will tell. Thanks for the library.
